attestation-container/attestation-container.go

```diff
@@ -38,27 +40,33 @@ func (s *server) FetchAttestation(ctx context.Context, in *pb.FetchAttestationRe
 		return nil, status.Errorf(codes.Internal, "failed to fetch attestation report: %s", err)
 	}
 
-	reportedTCBBytes := reportBytes[attest.REPORTED_TCB_OFFSET : attest.REPORTED_TCB_OFFSET+attest.REPORTED_TCB_SIZE]
-	chipIDBytes := reportBytes[attest.CHIP_ID_OFFSET : attest.CHIP_ID_OFFSET+attest.CHIP_ID_SIZE]
-	endorsement, err := attest.FetchAttestationEndorsement(*endorsementServer, reportedTCBBytes, chipIDBytes)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to fetch attestation endorsement: %s", err)
+	var endorsement []byte
+	if endorsementEnvironmentValue == nil {
```

Comment:
Considering the environment variable's value should be the same in the container's lifetime, caching the env var looks a nice idea. How about caching response from the server then?
If containers lifetime is not considerably long, is it fine to cache it?
If so, we can have `endorsement []byte` global var rather than `endorsementEnvironmentValue` and main would look like:
```go
func main()  {
// same as before

// fetch attestation endorsement
if fetchFromServer { endorsement = fetchFromServerAsBefore() }
else { endorsement = fetchFromEnvVar() }

// same as before
}
```

Reply:
After discussion, we are not 100% sure about the container's lifetime at this moment. We can leave it as it is.